src/cf.data.pre

```diff
@@ -10457,4 +10457,99 @@ DOC_START
 		server_pconn_for_nonretriable allow SpeedIsWorthTheRisk
 DOC_END
 
+NAME: happy_eyeballs_connect_timeout
```

Comment:
This directive is duplicate with the documented purpose of connect_timeout, and use of slow-setup connections makes the current value of that timeout effectively pointless.

IMO this new algorithm should use connect_timeout. Just reduce its default value instead of adding yet another timeout directive to the connect process.

Reply:
> This directive is duplicate with the documented purpose of connect_timeout

I see no duplication:

* connect_timeout limits the time spent on each connect(2) attempt
* happy_eyeballs_connect_timeout limits prime track's _exclusive_ access to connection establishment; it does not limit individual connect(2) attempts

By default, connect_timeout (one minute) is larger than happy_eyeballs_connect_timeout (250 milliseconds) so a spare track may start its attempts while the primary track is still on its first (slow) connection attempt. This is the desired behavior for many environments.

Setting connect_timeout to be smaller than happy_eyeballs_connect_timeout would limit prime track's exclusive access time to that smaller connect_timeout value (because a failed prime connect will trigger the start of the spare track), but I see nothing wrong with that per se.

> use of slow-setup connections makes the current value of that timeout effectively pointless.

I do not understand what problem you are describing -- default happy_eyeballs_connect_timeout works exactly as intended, delaying spare track attempts by 250 milliseconds if the primary connection is slow to establish. If the explanations and examples above did not address your concern, then please rephrase or detail the problem you see.